server/src/organize-imports.ts

```diff
@@ -8,12 +8,20 @@
 import * as lsp from 'vscode-languageserver';
 import * as tsp from 'typescript/lib/protocol';
 import { Commands } from './commands';
+import { CodeActionKind } from "vscode-languageserver";
 
-export function provideOrganizeImports(file: string, context: lsp.CodeActionContext, result: (lsp.Command | lsp.CodeAction)[]): void {
-    if (!context.only || context.only.indexOf(lsp.CodeActionKind.SourceOrganizeImports) === -1) {
+export function *provideOrganizeImports(
```

Comment:
Why have you switched to generators? Is there any benefit in that?

I think it would only make sense if the calling code does something with the result before the full result is obtained. In the case of getting diagnostics, the client needs the full list anyway so a simple loop seems like will be more efficient than generators which are essentially separate function calls to get each and every item.

Reply:
> I also refactored to use generators when processing tsserver responses instead of passing an array in and directly mutating it, which makes these functions easier to test and work with, in my opinion.

I can return a generated array, I mostly didn't like the passing of a mutable array in pattern